homeassistant/components/totalconnect/__init__.py

```diff
@@ -23,39 +20,32 @@
 TOTALCONNECT_PLATFORMS = ['alarm_control_panel']
 
 
-class TotalConnectSystem:
-    """TotalConnect System class."""
+def setup(hass, config):
+    """Set up TotalConnect component."""
+    conf = config[DOMAIN]
 
-    def __init__(self, username, password):
-        """Initialize the TotalConnect system."""
-        from total_connect_client import TotalConnectClient
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
 
-        _LOGGER.debug("Setting up TotalConnectSystem...")
-        self._username = username
-        self._password = password
-        self._client = TotalConnectClient.TotalConnectClient(username,
-                                                             password)
+    client = TotalConnectClient.TotalConnectClient(username, password)
 
-    @property
-    def client(self):
-        """Return the client."""
-        return self._client
+    if client.token is False:
+        return False
 
+    hass.data[DOMAIN] = TotalConnectSystem(username, password, client)
 
-def setup(hass, config):
-    """Set up TotalConnect component."""
-    conf = config.get(DOMAIN)
-    if conf is not None:
+    for platform in TOTALCONNECT_PLATFORMS:
+        discovery.load_platform(hass, platform, DOMAIN, {}, config)
 
-        username = conf.get(CONF_USERNAME)
-        password = conf.get(CONF_PASSWORD)
+    return True
 
-        hass.data[DOMAIN] = TotalConnectSystem(username, password)
 
-        for platform in TOTALCONNECT_PLATFORMS:
-            discovery.load_platform(hass, platform, DOMAIN, {}, config)
+class TotalConnectSystem:
+    """TotalConnect System class."""
 
-        return True
+    def __init__(self, username, password, client):
+        """Initialize the TotalConnect system."""
 
-    _LOGGER.critical("TotalConnect configuration is missing.")
-    return False
+        self._username = username
+        self._password = password
+        self._client = client
```

Comment:
Make the attribute non private by not starting with underscore.

Reply:
done